typescript/cli/src/registry/MergedRegistry.ts

```diff
@@ -0,0 +1,132 @@
+import { Logger } from 'pino';
+
+import {
+  BaseRegistry,
+  ChainAddresses,
+  GithubRegistry,
+  IRegistry,
+  RegistryContent,
+  RegistryType,
+} from '@hyperlane-xyz/registry';
+import { LocalRegistry } from '@hyperlane-xyz/registry/local';
+import { ChainMap, ChainMetadata, ChainName } from '@hyperlane-xyz/sdk';
+import {
+  isHttpsUrl,
+  objKeys,
+  objMerge,
+  rootLogger,
+} from '@hyperlane-xyz/utils';
+
+export interface MergedRegistryOptions {
+  primaryRegistryUri: string;
+  overrideRegistryUri: string;
+  isDryRun?: boolean;
+  logger?: Logger;
+}
+
+export class MergedRegistry extends BaseRegistry implements IRegistry {
+  public readonly type = RegistryType.Local;
+  public readonly primaryRegistry: IRegistry;
+  public readonly overrideRegistry: IRegistry;
+  public readonly isDryRun: boolean;
+
+  constructor({
+    primaryRegistryUri,
+    overrideRegistryUri,
+    logger,
+    isDryRun,
+  }: MergedRegistryOptions) {
+    logger ||= rootLogger.child({ module: 'MergedRegistry' });
+    super({ logger });
+
+    // If not provided, allow the GithubRegistry to use its default
+    if (isHttpsUrl(primaryRegistryUri)) {
+      this.primaryRegistry = new GithubRegistry({
+        uri: primaryRegistryUri,
+        logger: logger.child({ registry: 'primary-github' }),
+      });
+    } else {
+      this.primaryRegistry = new LocalRegistry({
+        uri: primaryRegistryUri,
+        logger: logger.child({ registry: 'primary-local' }),
+      });
+    }
+
+    this.overrideRegistry = new LocalRegistry({
+      uri: overrideRegistryUri,
+      logger: logger.child({ registry: 'override-local' }),
+    });
```

Comment:
why have this in the constructor?
I was expecting this to be a very thin wrapper over `registries: IRegistry[]` and just `fold(objMerge())` over them